property_tests/are_equal/opt.ts

```diff
@@ -0,0 +1,36 @@
+export function createAreOptsEqualCodeDeclaration(): string {
```

Comment:
Can we just use fast-equals inside of the Canister? Otherwise I'm thinking we should have a general-purpose object equality function that Opt, Record, Variant, etc can just use. And we should get rid of the nested function.

Reply:
Documented on issue #1390, will tackle later.